src/App.jsx

```diff
@@ -1,21 +1,34 @@
 import "./App.css";
-import { useState } from "react";
+import React, { useState, useEffect } from "react";
 import TodoList from "./components/TodoList.jsx";
 import AddTodoForm from "./components/AddTodoForm.jsx";
 
 function App() {
-  const [todoList, setTodoList] = useState([]);
+  const useSemiPersistentState = (key, initialValue) => {
```

Comment:
The useSemiPersistentState hook should be moved out of the App component and placed above it.